### PR TITLE
GridOut::write_mesh_per_processor_as_vtu() for simplex

### PR DESCRIPTION
**Pull request by peterrum**
While trying to port `GridOut::write_mesh_per_processor_as_vtu()` to simplex meshes, I observed the following two issues:

- `DataOutBase::Patch::points_are_available = false` is not working yet:
https://github.com/dealii/dealii/blob/ac854ef67344a07a72617e8def997c8fe9717eed/source/grid/grid_out.cc#L3673
- for creating the pvtu record `build_patches()` is called without mapping (and without `DoFHandler` attached so that the mapping index cannot be queried from the cell): 
https://github.com/dealii/dealii/blob/ac854ef67344a07a72617e8def997c8fe9717eed/source/grid/grid_out.cc#L3742

### CODE REVIEW

**Comment by peterrum** [comment]
`GridOut::write_mesh_per_processor_as_vtu()` works now also for simplex meshes. Ready once #11924 and #11925 are merged.

File: source/base/data_out_base.cc [hunk 1]
```diff
@@ -2631,17 +2631,33 @@ namespace DataOutBase
...
+                  Assert(
+                    patch.reference_cell != ReferenceCells::Pyramid,
+                    ExcMessage(
+                      "Pyramids need different ordering of the vertices, which is not implemented yet here."));
+
+                  node = patch.vertices[point_no];
+                }
+
+              return node;
+            };
```
Comment (drwells) [inline source/base/data_out_base.cc:2656]: I'd like to simplify the control flow here - it's hard to follow when we have a lambda named `compute_node` and a top-level function named `compute_node`.

Could we move the lambda into a new top-level function named `compute_simplex_node` and then rename `compute_node` to `compute_hypercube_node` or something similar? That would make this a lot easier to understand.

**Review by drwells — CHANGES_REQUESTED** [review]

File: source/base/data_out_base.cc (hunk 1 above)
Comment (peterrum) [reply source/base/data_out_base.cc]: Done!

**Review by drwells — APPROVED** [review]
